nrf-softdevice/src/temperature.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{raw, RawError, Softdevice};
+use defmt::info;
+use fixed::types::I30F2;
+
+#[derive(defmt::Format)]
+pub enum TempError {
+    Raw(RawError),
+}
+
+impl From<RawError> for TempError {
+    fn from(err: RawError) -> Self {
+        TempError::Raw(err)
+    }
+}
+
+/// Get temperature reading in Celcius
+///
+/// Note this blocks for ~50us
+pub fn temperature_celsius(_sd: &Softdevice) -> Result<I30F2, TempError> {
+    let mut temp: i32 = 0;
+    let ret = unsafe { raw::sd_temp_get(&mut temp) };
+    match RawError::convert(ret) {
```

Comment:
this can be made shorter:

```rust
RawError::convert(ret)?;
```

or if you want to print a warning


```rust
    RawError::convert(ret).dewarn(intern!("sd_temp_get"))?;
```